src/Storages/StorageFile.cpp

```diff
@@ -509,6 +497,27 @@ Pipe StorageFile::read(
             else
                 return metadata_snapshot->getColumns();
         };
+        // if we do multiple reads from pipe, we want to check if pipe is a regular file or a pipe
+        if (this_ptr->table_fd_was_used)
+        {
+            if (S_ISREG(this_ptr->table_fd))
```

Comment:
`S_ISREG` is a macro, that accept `stat.st_mode` not fd.

Reply:
Added `struct stat fd_stat` and fixed that.